### ISSUE TITLE
Due to the Sentinel community 1.8.6 version of Sentinel due to the null pointer exception thrown during logging the system protection rule invalidation bug. 由于Sentinel社区1.8.6版本的Sentinel由于出现记录日志时抛出空指针异常出现系统保护规则失效bug

### ISSUE THREAD

**Issue by bchengwang** [alibaba/spring-cloud-alibaba#3332]
我们鼓励使用英文，如果不能直接使用，可以使用翻译软件，您仍旧可以保留中文原文。另外请按照如下要求提交相关信息节省社区维护同学的理解成本，否则该讨论极有可能直接被忽视或关闭。
We recommend using English. If you are non-native English speaker, you can use the translation software. We recommend using English. If you are non-native English speaker, you can use the translation software. In addition, please submit relevant information according to the following requirements to save the understanding cost of community maintenances, otherwise the discussion is very likely to be ignored or closed directly.

**Which Component**
eg. Nacos Discovery, Sentinel
Sentinel
**Describe the bug**
由于Sentinel社区1.8.6版本的Sentinel出现系统保护规则失效。原因当满足SystemRule时抛出SystemBlockException被LogSlot的entry方法所捕获，在catch获取了SystemBlockException中的所触发的Rule `e.getRule().getId()` 导致抛出空指针异常。从而导致无法收到 SystemBlockException的异常通知，导致拦截失效。建议在 ` Entry entry = SphU.entry(resourceName, ResourceTypeConstants.COMMON_WEB, EntryType.IN);` 获取其中的 `entry.getBlockError()` 判断是否为空, 如果不为空那么就进行再次抛出 `entry.getBlockError()` 获取到的异常 的方式来做容错增强。从而解决由于LogSlot记录日志的抛出空指针异常导致无法收到SystemBlockException得bug以及其他相关问题。该问题涉及 使用1.8.6的Sentinel全部框架

Due to a system protection rule failure in Sentinel version 1.8.6, the system fails to intercept SystemBlockException when it is thrown upon satisfying the SystemRule. The issue arises because the LogSlot's entry method captures the SystemBlockException and retrieves the triggered Rule using `e.getRule().getId()`, which leads to a NullPointerException. As a result, the exception notification for SystemBlockException is not received, rendering the interception ineffective. It is recommended to modify the code by checking `entry.getBlockError()` after obtaining `Entry entry = SphU.entry(resourceName, ResourceTypeConstants.COMMON_WEB, EntryType.IN);`. If it is not null, re-throw the exception obtained from `entry.getBlockError()` to enhance fault tolerance. This solution addresses the bug caused by a NullPointerException when logging in LogSlot, which prevents the reception of SystemBlockException, along with other related issues. This problem affects the entire framework using Sentinel version 1.8.6.

**Simplest demo**
The URL of the simplest demo to reproduce the problem.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**
e.g.  MacOS  、Java8 、 Version 0.2.1.RELEASE
所使用版本 

```xml
<dependency>
    <groupId>com.alibaba.cloud</groupId>
    <artifactId>spring-cloud-alibaba-dependencies</artifactId>
    <version>2021.0.5.0</version>
    <type>pom</type>
    <scope>import</scope>
</dependency>
```

**Comment by ruansheng8** [comment]
This problem needs to be fixed by the `Sentinel` community (actually part of it has been fixed #2980 #2853), and I noticed that you have discussed this issue in the `Sentinel` community ([#3138](https://github.com/alibaba/Sentinel/issues/3138)), when the latest version of the `Sentinel` community is fixed for this problem, I will update the dependency version of `Sentinel` in `SCA`

**Comment by bchengwang** [comment]
> 这个问题需要社区修复`Sentinel`（实际上已经修复了一部分[#2980 ](https://github.com/alibaba/spring-cloud-alibaba/issues/2980) #2853），我注意到你在`Sentinel`社区（[#3138](https://github.com/alibaba/Sentinel/issues/3138)）讨论过这个问题，当社区最新版本`Sentinel`修复为这个问题，我会更新`Sentinel`in的依赖版本`SCA`

好的。不过我建议可以在以后版本可以做一些容错的判断避免由于sentinel的问题导致不可用
